src/test/java/net/library/controller/BookControllerTest.java

```diff
@@ -0,0 +1,1035 @@
+package net.library.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.library.model.entity.Book;
+import net.library.model.entity.BookItem;
+import net.library.model.entity.Genre;
+import net.library.model.entity.User;
+import net.library.model.request.GenreRequest;
+import net.library.repository.BookItemRepository;
+import net.library.repository.BookRepository;
+import net.library.repository.GenreRepository;
+import net.library.repository.UserRepository;
+import net.library.repository.enums.BookItemStatus;
+import net.library.service.BookService;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.util.UUID;
+
+import static net.library.util.HttpUtil.*;
+import static net.library.util.Utils.getUUID;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@ActiveProfiles("test")
+@SpringBootTest
+@AutoConfigureMockMvc
+class BookControllerTest {
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private BookService bookService;
+    @Autowired
+    private GenreRepository genreRepository;
+    @Autowired
+    private BookRepository bookRepository;
+    @Autowired
+    private BookItemRepository bookItemRepository;
+    @Autowired
+    private UserRepository userRepository;
+
+    @AfterEach
+    void cleanAfter() {
+        bookService.deleteAll();
+        userRepository.deleteAll();
+    }
+
+    @Test
+    void addBookSuccess() throws Exception {
```

Comment:
not a complete test. you should check that the object was really added to the db